framework/db/conditions/LikeConditionBuilder.php

```diff
@@ -78,6 +78,9 @@ public function build(ExpressionInterface $expression, array &$params = [])
             if ($value instanceof ExpressionInterface) {
                 $phName = $this->queryBuilder->buildExpression($value, $params);
             } else {
+                if(empty($value)){
```

Comment:
This is BC break - e.g. will skip properly supplied `0`. 
Should check for non-empty string e.g. `trim($value) !== ""`.
Actually whole block should be optimized.

Reply:
when I use your solution , then receive this error

    "name": "PHP Deprecated Warning",
    "message": "trim(): Passing null to parameter #1 ($string) of type string is deprecated",

Reply:
You are right, casting seems needed in any way, e.g. `trim((string)$value) !== ""`.
(how nicely it worked before 8.1)

Reply:
@metroo Why would you have `null` in this scenario? `LikeCondition` accepts only string or array of strings. Casting `null` to empty string may hide some errors, as you should not really have `null` here.

Reply:
@metroo Could you please show code example how do you get this error / reproduce error? 
[Seems to me](https://github.com/yiisoft/yii2/blob/57ae8aa995805950616589273488a6af2fd5746d/framework/db/conditions/LikeConditionBuilder.php#L61) that passing in non-empty array values could get through ..

Reply:
when you set array with  one argument as null value
   $values = array(0=>null);

Reply:
> when you set array with one argument as null value
> $values = array(0=>null);

And what is expected SQL query in that case? `field LIKE "%%"`?